### PR TITLE
add masthead selector mask

### PR DESCRIPTION
**Pull request by afebbraro**
## What does this PR do?
Add's a mask when the selector dropdown is open. Updates dropdown header padding.

## Please check off completed items as you work.
If a checklist item or section does not apply to your PR
then please remove it.
### Documentation
 - [x] Update Spark Docs Angular
 - [x] Update Spark Docs Vanilla
 - [x] Update Component Sass Var/Class Modifier table

### Code
 - [x] Build Component in Spark Vanilla (Sass, HTML, JS)
 - [x] Build Component in Spark Angular
 - [x] Unit Testing in Spark Vanilla (100% coverage, 100% passing)
- [x] Unit Testing in Spark Angular (100% coverage, 100% passing)

screen shot 2019-01-10 at 1 08 18 pm

screen shot 2019-01-10 at 11 10 12 am

### Accessibility
- [x] New changes abide by [accessibility requirements](https://sparkdesignsystem.com/docs/accessibility)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome (Mobile)
  - [x] Apple Safari (Mobile)
  - [x] Google Chrome
  - [x] Mozilla Firefox
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari

### Design Review
 - [x] Design reviewed and approved

### Screenshots
Add screenshots to help explain your PR if you'd like. However, this is not
expected.

### CODE REVIEW

File: packages/spark-core/tests/masthead.tests.js [hunk 1]
```diff
@@ -251,3 +381,66 @@ describe('focus trap tests', () => {
...
+  afterEach(() => {
+    document.body.innerHTML = '';
+    nav.addEventListener.restore();
+    iconContainer.addEventListener.restore();
+  });
+
+  it('should not add event listener to selector if it does not exist', () => {
+    const selectorTrigger = mastheadDiv.querySelector('[data-sprk-dropdown-trigger="dropdown-selector"]');
+    expect(selectorTrigger).eql(null);
+    // expect(iconContainer.addEventListener.getCall(0).args[0]).eql('click');
```
Comment (RCopeland) [inline packages/spark-core/tests/masthead.tests.js]: should remove the commented out code

File: src/angular/projects/spark-core-angular/src/lib/components/sprk-dropdown/sprk-dropdown.component.ts [hunk 2]
```diff
@@ -91,6 +124,8 @@ export class SparkDropdownComponent {
   @Input()
   title: string;
   @Input()
+  selector: string;
```
Comment (RCopeland) [inline src/angular/projects/spark-core-angular/src/lib/components/sprk-dropdown/sprk-dropdown.component.ts]: I see there's an Input selector and also a value for dropdownType 'mastheadSelector', do we need both?

Comment (afebbraro) [reply src/angular/projects/spark-core-angular/src/lib/components/sprk-dropdown/sprk-dropdown.component.ts]: @RCopeland the selector input string is for the masthead selector title. So we need it.

File: packages/spark-core/tests/masthead.tests.js (hunk 1 above)
Comment (afebbraro) [reply packages/spark-core/tests/masthead.tests.js]: done! thanks @RCopeland

**Pull request merged**